securedrop/tests/test_journalist.py

```diff
@@ -104,15 +105,36 @@ def test_unauthorized_access_redirects_to_login(self):
         resp = self.client.get(url_for('index'))
         self.assertRedirects(resp, url_for('login'))
 
-    def test_invalid_credentials(self):
+    def test_login_throttle(self):
+        db.LOGIN_HARDENING = True
+        try:
+            for _ in range(Journalist._MAX_LOGIN_ATTEMPTS_PER_PERIOD):
+                resp = self.client.post(url_for('login'),
+                                        data=dict(username=self.user.username,
+                                                  password='invalid',
+                                                  token='mocked'))
+                self.assert200(resp)
+                self.assertIn("Login failed", resp.data)
+
+            resp = self.client.post(url_for('login'),
+                                    data=dict(username=self.user.username,
+                                              password='invalid',
+                                              token='mocked'))
+            self.assert200(resp)
+            self.assertIn("Please wait at least {} seconds".format(
+                Journalist._LOGIN_ATTEMPT_PERIOD), resp.data)
+        finally:
+            db.LOGIN_HARDENING = False
+
+    def test_login_invalid_credentials(self):
         resp = self.client.post(url_for('login'),
                                 data=dict(username=self.user.username,
                                           password='invalid',
                                           token='mocked'))
         self.assert200(resp)
         self.assertIn("Login failed", resp.data)
 
-    def test_valid_credentials(self):
+    def test_login_valid_credentials(self):
```

Comment:
Please remove this. Just kidding, I'm the guiltiest person here in terms of sneaking in unrelated fixes. It's totally fine to stay.